src/main/java/ladder/controller/LadderController.java

```diff
@@ -37,20 +54,37 @@ private Participants getParticipants() {
     }
 
     private Ladder getLadder(int participantsCount) {
-        String height = getHeight();
+        int height = inputView.getHeight();
         return new Ladder(height, participantsCount, rungGenerator);
     }
 
-    private String getHeight() {
-        return inputView.getHeight();
+    private Prizes getPrizes(int participantsCount) {
+        List<String> prizeNames = inputView.getPrizeNames();
+        return new Prizes(prizeNames, participantsCount);
     }
 
-    private <T> T repeatUntilValid(Supplier<T> function) {
-        try {
-            return function.get();
-        } catch (IllegalArgumentException e) {
-            outputView.printError(e);
-            return repeatUntilValid(function);
+    private void printLadderResult(Participants participants, Prizes prizes, Ladder ladder) {
+        LadderResponseDto ladderResponseDto = ladder.getLadderResult();
+        outputView.printLadderResult(ladderResponseDto, participants.getNames(), prizes.getNames());
+    }
+
+    private String getName(GameResult gameResult) {
+        String name = inputView.getName();
+
+        if (name.equals(ALL_RESULT_INPUT_KEYWORD)) {
```

Comment:
NPE를 막는 습관으로 null이 아닌 것이 확실한 것 (상수 등)을 항상 앞에 두는 게 좋습니다
```suggestion
        if (ALL_RESULT_INPUT_KEYWORD.equals(name)) {
```
NPE를 막읍시다 NPE는 나의 원수

Reply:
오호 이런 사소한 꿀팁 너무 좋아합니다 ㅎㅎ
습관으로 새기겠습니다!! 😄